test/realm_test.dart

```diff
@@ -1002,7 +1003,99 @@ Future<void> main([List<String>? args]) async {
         var realm = Realm(config);
 
         final leak = realm.all<Dog>().changes.listen((data) {});
-        await Future<void>.delayed(Duration(milliseconds: 1));
+        await Future<void>.delayed(Duration(milliseconds: 20));
+        realm.close();
+      });
+
+      test('List notifications', () async {
+        var config = Configuration([Team.schema, Person.schema]);
+        var realm = Realm(config);
+
+        final team = Team('t1', players: [Person("p1")]);
+        realm.write(() => realm.add(team));
+
+        var firstCall = true;
+        final subscription = (team.players as RealmList<Person>).changes.listen((changes) {
+          if (firstCall) {
+            firstCall = false;
+            expect(changes.inserted.isEmpty, true);
```

Comment:
Not really related to this PR, but I got to admit I like `insertions`, etc. better. But I guess it is in the eye of the beholder.

Reply:
we use `changes` since it's something going to happen and `inserted` since it already happened when the event is called

Reply:
English is not my native tongue, but this doesn't sound right to me. To me `changes` is a description of what changed, and that may include `insertions` etc. Perhaps you could run it by a true native? Anyway, I have already approved the PR. I just noticed it differs from core and .NET, and I felt I preferred their wording.

Reply:
.NET uses `inserted`, `modified` etc.
Anyways I already added an agenda topic to discuss this.